### ISSUE TITLE
[Enhancement]: Record value of free shipping coupons for reporting

### ISSUE THREAD

**Issue by contendia** [woothemes/woocommerce#35892]
### Describe the solution you'd like

When setting up a free shipping coupon, it is typical to create it with a value of $0.00. When looking at the coupon report in Analytics > Coupons, the report shows the number of times the coupon has been used, but the total value of the coupon is always $0.00. However, this is not technically correct. The actual value of the coupon varies with each order, but it is essentially the amount that  the least expensive regular shipping option would have cost for each order if the coupon were not used. Meaning the total value of the coupon is never really zero. The total value should be the cumulative amount of the least expensive alternative shipping option for each order that used the coupon. 

Accounting needs accurate data on discounts, and showing free shipping coupons with a dollar value of zero is not accurate because the seller still actually pays for shipping. In other words, shipping is still a real expense for each of those orders. So each order that uses a free shipping coupon was actually discounted by a real value. 

Coupon reporting needs to record 'what would have been' the actual shipping cost for each order and return the sum of those as the total value of the coupons used.  

### Describe alternatives you've considered

According to support, zero value free shipping coupons are the way the system is meant to work. There is no workaround that I know of, and none was offered. For now, if staff creates the sale and enters the coupon, we manually record the shipping cost before entering the coupon. Then we tally that list to determine how much free shipping was given away. However, if a customer uses the coupon, we essentially have no way of knowing how much the shipping was discounted for that order.

**Comment by Ferdev** [comment]
I don't think we have a clear reporting definition yet. WooCommerce currently reports the discount applied by a coupon, while the merchant's shipping expense is a separate value, and the cheapest rate available to the customer is only one possible estimate of that expense. Calculating what shipping would have cost without the coupon would also require us to capture the alternative rates available at checkout, because rates, packages and addresses can change later. Before engineering starts, product and analytics should decide whether this metric means the customer's avoided shipping charge, the merchant's shipping cost, or another accounting value, and when we store it. I'd move this to a product decision first.
